File: app/src/main/java/com/mdgd/pokemon/ui/error/ErrorContract.java
package com.mdgd.pokemon.ui.error;

import com.mdgd.mvi.fragments.FragmentContract;

public class ErrorContract {

    public interface ViewModel extends FragmentContract.ViewModel<View> {
    }

    public interface View extends FragmentContract.View {
    }

    public interface Host extends FragmentContract.Host {
    }
}

File: app/src/main/java/com/mdgd/pokemon/ui/error/ErrorContract.kt
package com.mdgd.pokemon.ui.error

import com.mdgd.mvi.fragments.FragmentContract
import com.mdgd.pokemon.ui.error.state.ErrorFragmentState

class ErrorContract {
    interface ViewModel : FragmentContract.ViewModel<View, ErrorFragmentState>
    interface View : FragmentContract.View
    interface Host : FragmentContract.Host
}
